### ISSUE TITLE
Signature Help in Demo Server not working

### ISSUE THREAD

**Issue by rbasniak** [CXuesong/LanguageServer.NET#17]
Hello, first of all, great work you have done in this repo. It's very tempting to be able to write a LS in C#, I found many examples in Node, but C# is a lot more powerful.

I want to develop a LS for a new language and I'm playing with you Demo Language Server. In this demo you have a SignatureHelp that should be triggered by the `(` or `)` characters.

When I type them I get the following error in the demo VS Code extension:
```
[Error - 10:36:42] Request textDocument/signatureHelp failed.
  Message: Cannot find method "textDocument/signatureHelp" with matching signature.
  Code: -32602 
```
Then I went to the LSP specifications site and noticed that the demo is missing a parameter in the `SignatureHelp` method, so I changed it to this:

```
        [JsonRpcMethod]
        public SignatureHelp SignatureHelp(TextDocumentIdentifier textDocument, Position position, SignatureHelpContext context)
        {
            return new SignatureHelp(new List<SignatureInformation>
            {
                new SignatureInformation("**Function1**", "Documentation1"),
                new SignatureInformation("**Function2** <strong>test</strong>", "Documentation2"),
            });
        }
```

Now if I debug the LS and put a breakpoint in the `SignatureHelp` method, it hits the breakpoint. But nothing shows in VS Code.

Is there any extra step that I'm missing?

Thanks in advance,
Rodrigo Basniak

**Comment by CXuesong** [comment]
If you check the [JSON dump](https://github.com/CXuesong/LanguageServer.NET/blob/5c56b877b8abd0a5a8334bfeb99ff45f189ebfc0/DemoLanguageServer/Program.cs#L84) of the language server, you can see the JSON sent from / received at language server.

Example parameters payload
```json
{
  "textDocument": {
    "uri": ".../test.demo"
  },
  "position": {
    "line": 2,
    "character": 5
  },
  "context": {
    "isRetrigger": false,
    "triggerCharacter": "(",
    "triggerKind": 2
  }
}
```

Actual function signature:
https://github.com/CXuesong/LanguageServer.NET/blob/5c56b877b8abd0a5a8334bfeb99ff45f189ebfc0/DemoLanguageServer/Services/TextDocumentService.cs#L26

I think this is error caused by an extra parameter `context` introduced in later version of VSCode, when calling [`textDocument/signatureHelp`](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_signatureHelp). This extra parameter will fail the JSON RPC method resolution.

Then either adding this `context` parameter to `TextDocumentService.SignatureHelp` method, or specifying `AllowExtensionData = true` in ` [JsonRpcMethod]` attribute should resolve the bug.

I'll update the example with the former approach.

**Comment by CXuesong** [comment]
Fixed demo.

Also fixed a bug that could cause failure when LSP client processes the signature help response: c94dd55f718d3d656e41462a94c2610453782e3b

Released [0.3.6-int.2](https://www.nuget.org/packages/CXuesong.LanguageServer.VsCode/0.3.6-int.2).

**Comment by rbasniak** [comment]
Thank you for the time to update the demos and the new release.
